execution_engine/src/storage/trie/mod.rs

```diff
@@ -129,22 +133,44 @@ impl Default for PointerBlock {
 
 impl ToBytes for PointerBlock {
     fn to_bytes(&self) -> Result<Vec<u8>, bytesrepr::Error> {
-        self.0.to_bytes()
+        let mut result = bytesrepr::allocate_buffer(self)?;
+        for pointer in self.0.iter() {
+            result.append(&mut pointer.to_bytes()?);
+        }
+        Ok(result)
     }
 
     fn serialized_length(&self) -> usize {
-        self.0.serialized_length()
+        self.0.iter().map(ToBytes::serialized_length).sum()
     }
 }
 
 impl FromBytes for PointerBlock {
-    fn from_bytes(bytes: &[u8]) -> Result<(Self, &[u8]), bytesrepr::Error> {
-        FromBytes::from_bytes(bytes).map(|(arr, rem)| (PointerBlock(arr), rem))
+    fn from_bytes(mut bytes: &[u8]) -> Result<(Self, &[u8]), bytesrepr::Error> {
+        let pointer_block_array = {
+            let mut result: MaybeUninit<PointerBlockArray> = MaybeUninit::uninit();
```

Comment:
I'm having a hard time understanding this at first glance - would it be possible to add a comment that describes the justification for using `MaybeUninit`?